### PR TITLE
Feat/864: bargeIn and VAD at config level fix 

### PR DESCRIPTION
**Pull request by rammohan-y**
Fixed the issue of bargeIn with VAD at config level

### CODE REVIEW

**Comment by rammohan-y** [comment]
The following test cases are tested with this change

gather verb with minBargeinWordCount = 0 and vad defined: vad is enabled
gather verb with minBargeinWordCount = 0 and vad un-defined: vad is enabled with default sensitivity 2
gather verb with vad defined and minBargeinWordCount not defined: vad is enabled
gather verb with minBargeinWordCount > 0 : same as existing behavior

bargeIn config with minBargeinWordCount = 0 vad is not defined: vad is enabled with default sensitivity 2
bargeIn config with minBargeinWordCount > 0: same as existing behavior
In config only vad is defined and bargeIn config not defined: vad is enabled by setting minBargeinWordCount as 0

**Comment by vdharashive** [comment]
@davehorton @xquanluu can this be reviewed

File: lib/tasks/config.js [hunk 1]
```diff
@@ -295,9 +295,12 @@ class TaskConfig extends Task {
         voiceMs: this.vad.voiceMs || 250,
         silenceMs: this.vad.silenceMs || 150,
         strategy: this.vad.strategy || 'one-shot',
-        mode: this.vad.mod || 2
+        mode: this.vad.mode || 2
       };
     }
+    if (this.hasBargeIn && this.bargeIn.enable) {
+      cs.bargeIn = this.bargeIn;
```
Comment (xquanluu) [inline lib/tasks/config.js:302]: how do I reset bargeIn from call session level after bargeIn is set?

File: lib/tasks/gather.js [hunk 2]
```diff
@@ -125,6 +125,15 @@ class TaskGather extends SttTask {
   async exec(cs, {ep}) {
     this.logger.debug({options: this.data}, 'Gather:exec');
     await super.exec(cs, {ep});
+    if (this.data.minBargeinWordCount !== undefined) {
```
Comment (xquanluu) [inline lib/tasks/gather.js:128]: if (this.bargeIn.minBargeinWordCount)

is much better

**Comment by xquanluu** [comment]
Hi @rammohan-y 

https://github.com/jambonz/verb-specifications/blob/main/specs.json#L62

from this specification, the vad has its own properties to enable and disable, we cannot use minBargeinWordCount to enable or disable the vad, Could you please rethink to use vad object in config for your application.

This PR is invalid to me.

**Review by xquanluu — CHANGES_REQUESTED** [review]
invalid PR

**Pull request closed without merging**